test/SlowTests/Voron/Issues/RavenDB_10825.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Linq;
+using Sparrow;
+using Voron;
+using Voron.Impl;
+using Xunit;
+
+namespace SlowTests.Voron.Issues
+{
+    public class RavenDB_10825: StorageTest
+    {
+        protected override void Configure(StorageEnvironmentOptions options)
+        {
+            base.Configure(options);
+
+            options.MasterKey = Sodium.GenerateRandomBuffer((int)Sodium.crypto_aead_xchacha20poly1305_ietf_keybytes());
+            options.ManualFlushing = true;
+        }
+        
+        [Fact]
+        public unsafe void Encryption_buffer_of_freed_scratch_page_must_not_affect_another_overflow_allocation_on_tx_commit()
+        {
+            using (var tx = Env.WriteTransaction())
+            {
+                tx.LowLevelTransaction.State.NextPageNumber = 10;
+                
+                var page = tx.LowLevelTransaction.AllocatePage(1, 4);
+
+                Memory.Set(page.DataPointer, 10, 100);
+
+                tx.LowLevelTransaction.FreePage(4);
+
+                page = tx.LowLevelTransaction.AllocateOverflowRawPage(38888, out _, 3);
+                page.DataPointer[0] = 13;
+
+                var cryptoTxState = ((IPagerLevelTransactionState)tx.LowLevelTransaction).CryptoPagerTransactionState.Single().Value;
+
+                // explicitly change the order of items in dictionary so we'll apply the buffer of overflow page before the already freed one
```

Comment:
I'm not sure that this is something that you can _rely_ on, there is no guarantee of in order processing for dictionaries